features/FEATURE_LWIP/lwip-interface/lwipopts.h

```diff
@@ -96,7 +96,12 @@
 #endif
 
 #ifdef LWIP_DEBUG
+// For LWIP debug, double the stack
 #define TCPIP_THREAD_STACKSIZE      MBED_CONF_LWIP_TCPIP_THREAD_STACKSIZE*2
+#elif MBED_DEBUG
+// When debug is enabled on the build increase stack 25 percent, thread stack uses 8-byte alignment
+#define LWIP_ALIGN_UP(pos, align) ((pos) % (align) ? (pos) +  ((align) - (pos) % (align)) : (pos))
```

Comment:
If this alignment is needed here, isn't it potentially needed everywhere? Do they get a loud/clear error if they set something non-aligned in the JSON?

Reply:
Loud, maybe not so clear:
-------
[1528872011.15][CONN][RXD] ++ MbedOS Error Info ++
[1528872011.20][CONN][RXD] Error Status: 0x80ff0100 Code: 256 Module: 255
[1528872011.24][CONN][RXD] Error Message: Fatal Run-time error
[1528872011.26][CONN][RXD] Location: 0x12005
[1528872011.27][CONN][RXD] Error Value: 0x0
[1528872011.37][CONN][RXD] Current Thread: Id: 0x2000e8cc Entry: 0x12aad StackSize: 0x1000 StackMem: 0x2000e918 SP: 0x2000f7c0
[1528872011.40][CONN][RXD] -- MbedOS Error Info --
[1528872011.43][CONN][RXD] sys_thread_new create error
-------
We are using the low level rtos api:s which seem to not to protect from this. The standard  Thread.cpp would do the align for the behalf the the user.

Reply:
Okay, so maybe stick the same align macro on all the `#if` cases just in case while we're here.

Reply:
Ok, I added the macro to all stack size definitions.